packages/react-composites/src/composites/CallComposite/components/CallControls.tsx

```diff
@@ -86,8 +97,86 @@ export type CallControlOptions = {
    * @defaultValue true
    */
   screenShareButton?: boolean | { disabled: boolean };
+  /* @conditional-compile-remove-from(stable): custom button injection */
+  /**
+   * Inject custom buttons in the call controls.
+   *
+   * @beta
+   */
+  onFetchCustomButtonProps?: CustomCallControlButtonCallback[];
 };
 
+/* @conditional-compile-remove-from(stable): custom button injection */
+/**
+ * Placement for a custom button injected in the {@link CallControls}.
+ *
+ * 'first': Place the button on the left end (right end in rtl mode).
```

Comment:
If buttons like the options button is moved to the overflow menu, will this simply add the custom below the options button?

Reply:
At least the way I was thinking about it, this is just for the top-level button.

Good point about API implications.

We could either:
1. Reject the placement value (i.e. deprecation, but not backward compatible)
2. Move it below/above in the overflow menu

As you suggested, (2) is the API compatible change, although it'll still be a breaking change from a UX perspective. (1) might actually be better in that case -- it's better to require developer action than surprisingly move their button around.

Reply:
I think for allowing contoso to add new custom buttons we have to have an overflow menu, currently we have the max number of buttons we can fit on mobile so they would have to remove one to add a custom button to make use of this

Reply:
There is work planned for Contoso to be able to *explicitly* add buttons into the overflow menu. That will be a separate API. Contoso will be able to add a section to the overflow menu and add anything they want in there.

But we won't have any smarts about moving the button automagically between the two. That seems like surprising magic that is likely to create more confusion / trouble for the clients than be helpful.

Forking this discussion to team channel for more eyes.

Reply:
Resolving this based on discussion.

This API makes sense right now. Additionally we'll add API to add explicitly to the overflow menu.